### ISSUE TITLE
replace_consts false positive when using integer constant as part of range in match

### ISSUE THREAD

**Issue by 5225225** [rust-lang/rust-clippy#4969]
`cargo clippy -V`: `clippy 0.0.212 (69f99e7 2019-12-14)`

Playground link: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=401f208640cdf49300acf4ef902f01ee

Minimal example:

```rust
#![warn(clippy::replace_consts)]

fn main() {
    match 0_u32 {
        0 => "zero",
        // std::u32::MAX is suggested to be replaced with u32::max_value()
        1..=std::u32::MAX => "more",
    };
}
```

If I try replacing `std::u32::MAX` with `u32::max_value()`, I get a syntax error from rust. Wrapping in parenthesis or curly braces doesn't make it valid.

I don't get a warning if it's not part of a range.

**Comment by krishna-veerareddy** [comment]
Looks like the fix is to not lint `replace_consts` within patterns since function calls aren't allowed in them. I will tackle this one.
